onnx/test/shape_inference_test.py

```diff
@@ -67,7 +67,7 @@ def _make_graph(
                 )
                 input_value_infos.append(
                     make_tensor_value_info(
-                        "UNKNOWN_SHAPE_" + seed_name, TensorProto.INT64, ()
+                        "UNKNOWN_SHAPE_" + seed_name, TensorProto.INT64, (None,)
```

Comment:
Just a quick note on this. getShapeInput's rank inference [fails hard](https://github.com/onnx/onnx/blob/3dafb54b11a496cd933de17fd303324d0686fc3b/onnx/defs/shape_inference.cc#L475) when the shape is not 1 dimensional which makes this necessary.

Reply:
Maybe it would be worth modifying getShapeInput to avoid failing. Some operators takes scalar as inputs. Can it return something in case there is no dimension?

Reply:
Yes, I think that's quite reasonable. I had a more general question: is it clearly defined when we should `fail_shape_inference` and when we should simply not propagate shape information?

Reply:
@gramalingam would know better. If it does not fail, we could probably retrieve shape inference for the first nodes. Would users be interested into partial shape inference? That may be an interesting feature for another PR.

Reply:
It is common for inference to throw a shape-inference-error if an input is detected as having the wrong rank or shape. (The outer inference loop collects all such error messages, and tries to continue with inference for all nodes.)

I think it is okay for Reshape's shape-inference to fail if the input is detected as having the wrong rank. Are you running into any specific issue because of this?

Reply:
As far as I can see, the change you proposed (in old line 70, now moved to different line due to merged PR) to specify UNKNOWN_SHAPE's shape as `(None,)` seems correct. The older `()` seems incorrect.

Reply:
Well the only issue was that UNKNOWN_SHAPE was `()` but I think `(None,)` is what it should be. Given that, I'll let Reshape continue to fail hard if shape has an invalid rank. It should be ready for another look @gramalingam

Reply:
LGTM, thanks!